substrate/client/rpc-spec-v2/src/common/connections.rs

```diff
@@ -0,0 +1,60 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use jsonrpsee::ConnectionId;
+use parking_lot::Mutex;
+use std::{
+	collections::{HashMap, HashSet},
+	sync::Arc,
+};
+
+/// Limit the RPC functionality to a single connection.
+#[derive(Default, Clone)]
+pub struct RpcConnections {
+	data: Arc<Mutex<HashMap<ConnectionId, HashSet<String>>>>,
```

Comment:
could also be `Arc<Mutex<HashSet<(ConnectionId, SubscriptionId)>>>`  I guess but perhaps you have plan for it to more generic

Reply:
mmm since `ConnectionId` is just a small u64, I'd just combine to `Arc<Mutex<HashSet<(ConnectionId, SubscriptionId)>>>` and that hsould be more efficient and probably small or no real extra space cost :)